subworkflows/nf-core/gatk_align_and_preprocess/main.nf

```diff
@@ -0,0 +1,121 @@
+//
+// Performs GATK best practice alignment and pre-processing of reads using BWA, GATK mergebamalignments (where necessary), markduplicates, sortsam, samtools index and BQSR.
+// BWA index created from fasta file if not already provided
+//
+include { BWAMEM2_INDEX } from '../../../modules/bwamem2/index/main.nf'
+include { SAMTOOLS_FASTQ } from '../../../modules/samtools/fastq/main.nf'
+include { BWAMEM2_MEM   } from '../../../modules/bwamem2/mem/main.nf'
+include { GATK4_MERGEBAMALIGNMENT } from '../../../modules/gatk4/mergebamalignment/main.nf'
+include { PICARD_MARKDUPLICATES } from '../../../modules/picard/markduplicates/main.nf'
+include { PICARD_SORTSAM } from '../../../modules/picard/sortsam/main.nf'
+include { SAMTOOLS_INDEX } from '../../../modules/samtools/index/main.nf'
+include { GATK4_BASERECALIBRATOR } from '../../../modules/gatk4/baserecalibrator/main.nf'
+include { GATK4_APPLYBQSR } from '../../../modules/gatk4/applybqsr/main.nf'
+
+workflow GATK_ALIGN_AND_PREPROCESS {
+    take:
+    input       // channel: [ val(meta), [ input ], [ input_index ], [] ]
+    fasta               // channel: /path/to/reference/fasta
+    fai                 // channel: /path/to/reference/fasta/index
+    dict                // channel: /path/to/reference/fasta/dictionary
+    bwaindex            // channel: name for panel of normals
+    is_ubam
+    sort_order
+    intervals
+    knownsites
+    knownsites_tbi
+
+    main:
+    ch_versions      = Channel.empty()
+
+    //
+    //If no bwa index has been provided, use bwamem2 index to create one from the fasta file
+    //
+    if (bwaindex) {
+        ch_bwa_index = bwaindex
+    } else {
+        BWAMEM2_INDEX( fasta )
+        ch_bwa_index = BWAMEM2_INDEX.out.index
+        ch_versions = ch_versions.mix(BWAMEM2_INDEX.out.versions)
+    }
```

Comment:
I do think this should be done separately in a local `prepare_genome` subworkflow like in rnaseq